### DISCUSSION TITLE
Ability to reload single extension instead reloading entire window/all extensions.

### DISCUSSION THREAD

**Discussion by venkat-karnati** [microsoft/vscode-discussions#1531] [Extension Development QnA] [answered]
I'm developing a custom vs code extension and i want to develop a command/capability to reload custom extension alone. How to do this?

I don't want to use existing "Reload Window" or "Reload extension host" commands because it will reload all extensions.

**Comment by alefragnani** [comment]
Hi @venkat-karnati ,

The current Extensions API (https://code.visualstudio.com/api/references/vscode-api#extensions) does not provide capabilities to _control_ other extension. It only allows you to interact with it (via its exposed APIs), load basic info and activate it. But nothing like reload or stop. Neither the _command line interface_ provide options for that.

BTW, I’m not sure this would be against the proposals of the extension API. I mean, what’s the downside of allowing extension restarting other extensions? How this could be made in a way to protect extensions? But, even so, I would suggest you to open a feature request in the VS Code repo (https://github.com/microsoft/vscode/issues/new/choose), asking for such feature, so the team could properly evaluate.

Hope this helps

**Reply by venkat-karnati** [reply]
Hi @alefragnani 
I don't want to reload other extensions. I'm developing a custom extension and want to reload my own extension alone post any configuration changes in extension settings. Is it possible to achieve?

**Reply by alefragnani** [reply, marked as answer]
There is no need to reload the extension simply because some setting has changed. You just have to handle the `onDidChangeConfiguration` event and update your extension according the new settings.

Something like this:

```typescript
    context.subscriptions.push(vscode.workspace.onDidChangeConfiguration(cfg => {

        if (cfg.affectsConfiguration("workbench.iconTheme")) {
            providerManager.refreshTreeViews()
        }
    }));
```

A full sample is provided here https://github.com/microsoft/vscode-extension-samples/tree/main/configuration-sample

Hope this helps

**Reply by venkat-karnati** [reply]
Will check and revert back for any queries. Thanks for quick response @alefragnani
